beer-garden/functionary
# #75 - Added support for posgresql and sqlite databases, postgres is d…

Closes #75 

This PR adds the ability for users to use Postgresql as their database of choice, instead of sqlite. User's can switch between the two by updating the `DB_ENGINE` in the docker-compose.yml.

## Testing
- If using the vscode workspace, run the `functionary: docker compose up` task for deploying functionary with postgresql.
  - **Edit the `DB_ENGINE` in the docker-compose if you plan on using sqlite. Postgresql is the default.**  
- Verify the correct DB engine is being used by getting a shell into the `functionary-django` container, running `python manage.py shell`, and then running the following:
```python
from django.db import connection
print(connection.vendor)
```
- If not using the vscode workspace, reference the task commands to grab the docker-compose command for both tasks described above

## Implementation Notes
- The Postgresql container will always get deployed, regardless of the value of `DB_ENGINE`
- In the database service in the docker-compose.yml:
  - `DB_USER` should match `POSTGRES_USER`
  - `DB_PASSWORD` should match `POSTGRES_PASSWORD`
  - `DB_NAME` should match `POSTGRES_DB`
  - `DB_HOST` should match `hostname`
